packages/transactions/src/TransferHotspotV1.ts

```diff
@@ -0,0 +1,126 @@
+import proto from '@helium/proto'
+import Transaction from './Transaction'
+import {
+  EMPTY_SIGNATURE,
+  toUint8Array,
+  toAddressable,
+  toNumber,
+} from './utils'
+import { Addressable, SignableKeypair } from './types'
+
+interface TransferHotspotOptions {
+  gateway?: Addressable
+  seller?: Addressable
+  buyer?: Addressable
+  sellerSignature?: Uint8Array
+  buyerSignature?: Uint8Array
+  buyerNonce?: number
+  amountToSeller?: number
+  fee?: number
+}
+
+interface SignOptions {
+  seller?: SignableKeypair
+  buyer?: SignableKeypair
+}
+
+export default class TransferHotspotV1 extends Transaction {
+  public gateway?: Addressable
+
+  public seller?: Addressable
+
+  public buyer?: Addressable
+
+  public sellerSignature?: Uint8Array
+
+  public buyerSignature?: Uint8Array
+
+  public buyerNonce?: number
+
+  public amountToSeller?: number
+
+  public fee?: number
+
+  constructor(opts: TransferHotspotOptions) {
+    super()
+    this.gateway = opts.gateway
+    this.seller = opts.seller
+    this.buyer = opts.buyer
+    this.sellerSignature = opts.sellerSignature
```

Comment:
I think the signatures need to come after the `calculateFee()` line because that will null them out for the fee calculation. Also, in PaymentV2, I added a condition where if the fee is passed in, it just uses that and skips the fee calculation

Reply:
Makes sense, updating